deploy-board/deploy_board/webapp/cluster_view.py

```diff
@@ -1448,3 +1446,21 @@ def get(self, request, name, stage):
         }
 
         return render(request, 'clusters/base_image_history.html', data)
+
+def get_cluster_name(request, name, stage):
+    cluster_name = '{}-{}'.format(name, stage)
+    current_cluster = clusters_helper.get_cluster(request, cluster_name)
+    if current_cluster is None:
+        env = environs_helper.get_env_by_stage(request, name, stage)
+        cluster_name = env.get("clusterName")
+    return cluster_name
```

Comment:
do we need the check for `{}-{}` format cluster existing, or could we change this to 
```suggestion
def get_cluster_name(request, name, stage):
    env = environs_helper.get_env_by_stage(request, name, stage)
    cluster_name = env.get("clusterName")
    return cluster_name
```

Reply:
Not necessarily. I left it just to be extra careful. If we entirely depend on the env to lookup a cluster there could be scenario where the cluster name on the env does not exist or is a mismatch with the actual cluster or an API issue that prevents the lookup. Not likely, but still possible. If we don't have that link for any reason we still want to be able to get/update those clusters.